yb-voyager/cmd/migration_complexity.go

```diff
@@ -200,3 +210,121 @@ func getComplexityForLevel(level string, count int) string {
 		panic(fmt.Sprintf("unknown impact level %s for determining complexity", level))
 	}
 }
+
+// ======================================= Migration Complexity Explanation ==========================================
+
+const explainTemplateHTML = `
+<table border="1" cellpadding="5" cellspacing="0" style="border-collapse: collapse;">
+    <thead>
+        <tr>
+            <th>Category</th>
+            <th>Level-1</th>
+            <th>Level-2</th>
+            <th>Level-3</th>
+            <th>Total</th>
+        </tr>
+    </thead>
+    <tbody>
+    {{- range .Summaries }}
+        <tr>
+            <td>{{ .Category }}</td>
+            <td>{{ index .ImpactCounts "LEVEL_1" }}</td>
+            <td>{{ index .ImpactCounts "LEVEL_2" }}</td>
+            <td>{{ index .ImpactCounts "LEVEL_3" }}</td>
+            <td>{{ .TotalIssueCount }}</td>
+        </tr>
+    {{- end }}
+    </tbody>
+</table>
+
+<p>
+	Level-1 Impact: Resolutions are available with minimal effort.<br/>
+	Level-2 Impact: Resolutions are available requiring moderate effort.<br/>
+	Level-3 Impact: Resolutions may not be available or are highly complex.
+</p>
+
+<p>
+	<strong>Reasoning:</strong> {{ .ComplexityRationale }}
+</p>
+`
+
+const explainTemplateText = `Reasoning: {{ .ComplexityRationale }}`
+
+type ExplanationData struct {
+	Summaries           []CategorySummary
+	ComplexityRationale string // short reasoning or explanation text
+}
+
+type CategorySummary struct {
+	Category        string
+	TotalIssueCount int
+	ImpactCounts    map[string]int // e.g. {"Level-1": 3, "Level-2": 5, "Level-3": 2}
+}
+
+func buildMigrationComplexityExplanation(sourceDBType string, assessmentReport AssessmentReport, reportFormat string) (string, error) {
+	if sourceDBType != POSTGRESQL {
+		return "", nil
+	}
+
+	var explanation ExplanationData
+	explanation.ComplexityRationale = migrationComplexityRationale
+
+	explanation.Summaries = buildCategorySummary(assessmentReport.Issues)
+
+	var tmpl *template.Template
+	var err error
+	if reportFormat == "html" {
+		tmpl, err = template.New("Explain").Parse(explainTemplateHTML)
+	} else {
+		tmpl, err = template.New("Explain").Parse(explainTemplateText)
+	}
+
```

Comment:
why are doing this here and not in HTML assessment template?

Reply:
we will have to add the explanation structure in the assessment report struct also.
This is also a low level detail which can reside separately...